### PR TITLE
sync tabs and focus after close a tab

### LINKED ISSUE #816: Close an unfocused tab (using mouse) results in focus switched
This bug comes with #737. Very sorry for not having a full test.

Repo:
open tab a,b,c,d
activate c
mouse close a
expect c is shown and is the only highlight in tabview
alt-1
expect b is shown and is the only highlight in tabview

### PR DESCRIPTION
**Pull request by bef178**
Should fix #816
Should go with #737

### CODE REVIEW

**Comment by bef178** [comment]
Manual test cases:

open tab a,b,c,d
activate a
ctrl-w
ctrl-w
expect c is shown and is the only highlight in tabview

open tab a,b,c,d
activate a
mouse close a
mouse close b
expect c is shown and is the only highlight in tabview

open tab a,b,c,d
activate c
mouse close a
expect c is shown and is the only highlight in tabview
alt-1
expect b is shown and is the only highlight in tabview

open tab a,b,c,d
activate c
mouse close a
expect c is shown and is the only highlight in tabview
mouse activate b
expect b is shown and is the only highlight in tabview

open tab a,b,c,d,e
activate c
ctrl-w
ctrl-w
ctrl-w
expect d is shown and is the only highlight in tabview
ctrl-w
expect e is shown and tabview is hided

File: src/cascadia/TerminalApp/App.cpp [hunk 1]
```diff
@@ -851,20 +851,38 @@ namespace winrt::TerminalApp::implementation
         {
             _lastTabClosedHandlers();
         }
+
         uint32_t tabIndexFromControl = 0;
         _tabView.Items().IndexOf(tabViewItem, tabIndexFromControl);
 
-        if (tabIndexFromControl == _GetFocusedTabIndex())
+        int focusedTabIndex = _GetFocusedTabIndex();
```
Comment (miniksa) [inline src/cascadia/TerminalApp/App.cpp]:
```suggestion
        const auto focusedTabIndex = _GetFocusedTabIndex();
```

File: src/cascadia/TerminalApp/App.cpp [hunk 2]
```diff
@@ -690,7 +690,7 @@ namespace winrt::TerminalApp::implementation
     //      Negative values of `delta` will move the view up by one page, and positive values
     //      will move the viewport down by one page.
     // Arguments:
-    // - delta: The direction to move the view relative to the current viewport(it 
+    // - delta: The direction to move the view relative to the current viewport(it
     //      is clamped between -1 and 1)
```
Comment (miniksa) [inline src/cascadia/TerminalApp/App.cpp]:
```suggestion
    //      (it is clamped between -1 and 1)
```

File: src/cascadia/TerminalApp/App.cpp [hunk 3]
```diff
@@ -690,7 +690,7 @@ namespace winrt::TerminalApp::implementation
     //      Negative values of `delta` will move the view up by one page, and positive values
     //      will move the viewport down by one page.
     // Arguments:
-    // - delta: The direction to move the view relative to the current viewport(it 
+    // - delta: The direction to move the view relative to the current viewport(it
```
Comment (miniksa) [inline src/cascadia/TerminalApp/App.cpp]:
```suggestion
    // - delta: The direction to move the view relative to the current viewport
```

File: src/cascadia/TerminalApp/App.cpp [hunk 4]
```diff
@@ -851,20 +851,38 @@ namespace winrt::TerminalApp::implementation
...
+            _tabView.SelectedIndex((tabIndexFromControl > 0) ? (tabIndexFromControl - 1) : 1);
         }
 
         // Removing the tab from the collection will destroy its control and disconnect its connection.
         _tabs.erase(_tabs.begin() + tabIndexFromControl);
         _tabView.Items().RemoveAt(tabIndexFromControl);
 
         // ensure tabs and focus is sync
-        _tabView.SelectedIndex(tabIndexFromControl > 0 ? tabIndexFromControl - 1 : 0);
+        // focused tab shifts to the left if possible
```
Comment (miniksa) [inline src/cascadia/TerminalApp/App.cpp]: I'd prefer more verbose comments on each wing of this if/elseif/else. 
It requires a lot of mental work to deduce what each branch of this is doing for someone who hasn't been rooting around in the tab indexes.

File: src/cascadia/TerminalApp/App.cpp (hunk 1 above)
Comment (bef178) [reply src/cascadia/TerminalApp/App.cpp]: Will update

File: src/cascadia/TerminalApp/App.cpp (hunk 2 above)
Comment (bef178) [reply src/cascadia/TerminalApp/App.cpp]: Actually it results from auto trimming of  trailing space. Not quite sure should i change unrelated lines, but i will follow your comment first.

File: src/cascadia/TerminalApp/App.cpp (hunk 4 above)
Comment (bef178) [reply src/cascadia/TerminalApp/App.cpp]: Will update

File: src/cascadia/TerminalApp/App.cpp (hunk 3 above)
Comment (bef178) [reply src/cascadia/TerminalApp/App.cpp]: (same as above)

**Comment by bef178** [comment]
Is there any thing need to be changed? @miniksa

**Comment by bef178** [comment]
Closing since the related issue #816 is closed.

**Comment by miniksa** [comment]
Sorry, @bef178, I get a ton of mail these days. I didn't get back to this until now. It looks like there was another pull that solved this.

**Pull request closed without merging**